app/models/census_lat_long.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'roo_helper/shared'
+
+class CensusLatLong
+  include RooHelper
+
+  # Creates a ZIP file of CSVs by combining results from
+  #  - add_institution_addresses
+  #  - add_weams_physical_addresses
+  #  - add_weams_mailing_addresses
+  def self.export
+    missing_lat_long_physical_weams = Weam.missing_lat_long_physical
+    missing_lat_long_mailing_weams = Weam.missing_lat_long_mailing
+
+    weams_physical_facility_codes = missing_lat_long_physical_weams.map(&:facility_code).uniq
+    weams_mailing_facility_codes = missing_lat_long_mailing_weams.map(&:facility_code).uniq
+    weams_facility_codes = weams_physical_facility_codes + weams_mailing_facility_codes.uniq
+
+    addresses = []
+    add_institution_addresses(addresses, weams_facility_codes)
+    add_weams_physical_addresses(addresses, missing_lat_long_physical_weams)
+    add_weams_mailing_addresses(addresses, missing_lat_long_mailing_weams, weams_physical_facility_codes)
+
+    csvs = []
+    addresses.compact.in_groups_of(10_000, false) do |batch|
+      csvs << generate_csv(batch)
+    end
+
+    Group.export_csvs_as_zip(csvs, name)
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # Adds Approved Institutions rows from latest version that do not have a latitude or longitude
+  #   - if the facility code is present in weams_facility_codes it is ignored
+  def self.add_institution_addresses(addresses, weams_facility_codes)
+    Institution.missing_lat_long(Version.latest).each do |institution|
+      next if weams_facility_codes.include?(institution.facility_code)
+
+      value = [institution.physical_address || institution.address,
+               institution.physical_city || institution.city,
+               institution.physical_state || institution.state,
+               institution.physical_zip || institution.zip]
```

Comment:
I think we want all of physical or all of mailing not a possible mix of both

Reply:
changed